package_test.go

```diff
@@ -159,31 +176,40 @@ echo $token
 
 					conjur.AddSecret(variableIdentifier, secretValue)
 
-					out, err := exec.Command(PackageName, variableIdentifier).Output()
+					stdout, _, err := RunCommand(PackageName, variableIdentifier)
 
 					So(err, ShouldBeNil)
-					So(string(out), ShouldEqual, secretValue)
+					So(stdout.String(), ShouldEqual, secretValue)
 				})
 
 				Convey("Returns 404 on non-existent variable", func() {
 					variableIdentifier := "non-existent-variable"
 
-					out, err := exec.Command(PackageName, variableIdentifier).Output()
+					_, stderr, err := RunCommand(PackageName, variableIdentifier)
 
 					So(err, ShouldNotBeNil)
-					So(string(out), ShouldContainSubstring, "404")
+					So(stderr.String(), ShouldContainSubstring, "404")
 				})
 
 				Convey("Given a non-existent TokenFile is set", func() {
-					os.Setenv("CONJUR_AUTHN_TOKEN_FILE", "non-existent-user")
-
-					Convey("Returns with timed out error", func() {
-						variableIdentifier := "existent-or-non-existent-variable"
-
-						out, err := exec.Command(PackageName, variableIdentifier).Output()
-
-						So(err, ShouldNotBeNil)
-						So(string(out), ShouldContainSubstring, "timed out")
+					os.Setenv("CONJUR_AUTHN_TOKEN_FILE", "non-existent-token-file")
+
+					Convey("Waits for longer than a second", func() {
+						timeout := time.After(1 * time.Second)
+						unexpected_response := make(chan int)
+
+						go func() {
+							variableIdentifier := "existent-or-non-existent-variable"
+							RunCommand(PackageName, variableIdentifier)
+							unexpected_response <- 1
+						}()
+
+						select {
+						case <- unexpected_response:
+							So("receive unexpected response", ShouldEqual, "not receive unexpected response")
+						case <- timeout:
+							So(true, ShouldEqual, true)
+						}
```

Comment:
I'm confused as to what's happening here and what's the intent. Would it be possible to make the code a bit clearer?